source/ShopPanel.cpp

```diff
@@ -562,29 +430,113 @@ bool ShopPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
 		activePane = (activePane == ShopPane::Main ? ShopPane::Sidebar : ShopPane::Main);
 	else if(key == 'f')
 		GetUI()->Push(new Dialog(this, &ShopPanel::DoFind, "Search for:"));
-	else if(key == 'b' || key == 's' || key == 'r' || key == 'c' || key == 'i' || key == 'u')
+	else if(key == 'b')
 	{
-		// Handle Transactions:
-		// Outfitter: Buy, Sell, Uninstall, Install, [Buy or Move to] Cargo, Store.
-		// Shipyard: Buy, Sell ship and outfits, Sell ship and retain outfits in storage.
-		const auto result = CanDoTransaction(key);
+		// Outfitter: Buy and install up to <modifier> outfits for each selected ship.
+		// Shipyard: Buy up to <modifier> ships.
+		result = CanDoBuyButton();
 		if(result)
+			DoBuyButton();
+	}
+	else if(key == 's')
+	{
+		// In the Shipyard: Sell selected ships and outfits.
+		// In the Outfitter: Sell <modifier> of the selected outfit from each selected ship.
+		if(isOutfitter)
 		{
-			DoTransaction(key);
-
-			// Ship-based updates to cargo are handled when leaving.
-			// Ship-based selection changes are asynchronous, and handled by ShipyardPanel.
-			if(isOutfitter)
+			result = CanSellOrUninstall("sell");
+		}
+		else
+		{
+			result = playerShip;
```

Comment:
CanSell in the Shipyard context was removed as it was just "return playerShip;" and since I removed CanSell from the OutfitterPanel, I took it out of all the ShopPanel derivatives and applied the logic here instead.  CanSellOrUninstall was promoted from OutfitterPanel to the ShopPanel so that the middle layer of CanSell/CanUninstall/CanMoveToStorage could all be removed.